src/main/java/core/basesyntax/operation/SupplyOperation.java

```diff
@@ -0,0 +1,13 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void processOperation(FruitTransaction fruitTransaction) {
+        int fruitsInStorage = Storage.storage.get(fruitTransaction.getFruit());
```

Comment:
This line may cause a NullPointerException if the fruit is not already in the storage. Consider using `getOrDefault` method to provide a default value of 0 if the fruit is not found.